### PR TITLE
Create monotonic_binary_search.cpp

### PR DESCRIPTION
**Pull request by hazel11182**
This will help solve problems even if the array is not given as we use the concept of monotonic functions. Helps us acheive logn complexity instead of the brute force approach.

### CODE REVIEW

**Review by TawfikYasser — CHANGES_REQUESTED** [review]
I think that your code not working good

After the user enter the input `n` noting happens

Please check it again

File: searching/c-or-cpp/monotonic_binary_search.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+using namespace std;
+int square_root(int a,int p){
+    int s=0;
+    int e=a;
+    int ans=-1;
```
Comment (hazel11182) [inline searching/c-or-cpp/monotonic_binary_search.cpp:6]: needs to be float ans.

Comment (TawfikYasser) [reply searching/c-or-cpp/monotonic_binary_search.cpp:6]: Okay you can change it and re-commit

**Review by hazel11182 — COMMENTED** [review]
two changes need to be done
1. it has to be float square_root
2. it has to be float ans=-1

**Comment by TawfikYasser** [comment]
> two changes need to be done
> 
> 1. it has to be float square_root
> 2. it has to be float ans=-1

Okay no problem, You can start fixing the code

**Comment by TawfikYasser** [comment]
You do not need to create new Pull Request

**Comment by hazel11182** [comment]
This was my very first contribution, sorry for creating a new pull request.
what do i do now?

**Comment by TawfikYasser** [comment]
> This was my very first contribution, sorry for creating a new pull request.
> what do i do now?

No problem :) , You can close this pull request and continue with the [new](https://github.com/MakeContributions/DSA/pull/114)

**Pull request closed without merging**